library/lms.c

```diff
@@ -499,13 +498,20 @@ static int get_merkle_path( mbedtls_lms_private_t *ctx,
                             unsigned int leaf_node_id,
                             unsigned char *path )
 {
-    unsigned char tree[MERKLE_TREE_NODE_AM_MAX][MBEDTLS_LMS_M_NODE_BYTES_MAX];
     unsigned int curr_node_id = leaf_node_id;
     unsigned int adjacent_node_id;
     unsigned int height;
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+    unsigned char *tree = NULL;
 
-    ret = calculate_merkle_tree( ctx, ( unsigned char * )tree );
+    tree = mbedtls_calloc( MERKLE_TREE_NODE_AM(ctx->params.type),
+        MBEDTLS_LMS_M_NODE_BYTES(ctx->params.type) );
+    if ( !tree )
```

Comment:
Style: for readability, we avoid implicit truthiness of integers and pointers.
```suggestion
    if ( tree == NULL )
```

Also goes in `mbedtls_lms_calculate_public_key`.

(Note: there are violations of our current style rules around parentheses and whitespace. We're going to change those rules soon, so I'm not going to insist on the current rules.)

Reply:
Done